### DISCUSSION TITLE
appDir: how did you avoid hard navigation on app\[lang]?

### DISCUSSION THREAD

**Discussion by Marsony** [vercel/next.js#49858] [Help]
### Summary

Hello everyone

So, opened [this issue](https://github.com/vercel/next.js/issues/49799).

1. wrap your `app` in `[lang]`
2. disable `prefetch` on all of the `next\link`
3. click on the link
4. you got php-like hard navigation with window.reload

Is it possible to make a `soft` navigation with `i18n emulation (app\[lang])`?
Nextjs down i18n support on `appDir` which we had on the `pages`
Ty

### Example

[Here](https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing)

**Comment by holicreact** [comment]
@Marsony 나도 궁금해!
